rust/arrow/src/json/reader.rs

```diff
@@ -888,265 +855,362 @@ impl Decoder {
+            }
+            DataType::Float32 => self.read_primitive_list_values::<Float32Type>(rows),
+            DataType::Float64 => self.read_primitive_list_values::<Float64Type>(rows),
+            DataType::Timestamp(_, _)
+            | DataType::Date32(_)
+            | DataType::Date64(_)
+            | DataType::Time32(_)
+            | DataType::Time64(_) => {
+                return Err(ArrowError::JsonError(
+                    "Temporal types are not yet supported, see ARROW-4803".to_string(),
+                ))
+            }
+            DataType::Utf8 => {
+                StringArray::from_iter(flatten_json_string_values(rows).into_iter())
+                    .data()
+            }
+            DataType::LargeUtf8 => {
+                LargeStringArray::from_iter(flatten_json_string_values(rows).into_iter())
+                    .data()
+            }
+            DataType::List(field) => {
+                let child = self
+                    .build_nested_list_array::<i32>(&flatten_json_values(rows), field)?;
+                child.data()
+            }
+            DataType::LargeList(field) => {
+                let child = self
+                    .build_nested_list_array::<i64>(&flatten_json_values(rows), field)?;
+                child.data()
+            }
+            DataType::Struct(fields) => {
+                // extract list values, with non-lists converted to Value::Null
+                let len = rows.len();
+                let num_bytes = bit_util::ceil(len, 8);
+                let mut null_buffer =
+                    MutableBuffer::new(num_bytes).with_bitset(num_bytes, false);
+                let mut struct_index = 0;
+                let rows: Vec<Value> = rows
+                    .iter()
+                    .map(|row| {
+                        if let Value::Array(values) = row {
+                            values.iter().for_each(|_| {
+                                bit_util::set_bit(null_buffer.data_mut(), struct_index);
+                                struct_index += 1;
+                            });
+                            values.clone()
+                        } else {
+                            struct_index += 1;
+                            vec![Value::Null]
+                        }
+                    })
+                    .flatten()
+                    .collect();
+                let arrays =
+                    self.build_struct_array(rows.as_slice(), fields.as_slice(), &[])?;
+                let data_type = DataType::Struct(fields.clone());
+                let buf = null_buffer.freeze();
+                ArrayDataBuilder::new(data_type)
+                    .len(rows.len())
+                    .null_bit_buffer(buf)
+                    .child_data(arrays.into_iter().map(|a| a.data()).collect())
+                    .build()
+            }
+            DataType::Dictionary(_, _) => {
+                todo!()
```

Comment:
Is this planned in this PR? Otherwise I think it may make sense to create an error with a link to a Arrow issue?

Reply:
I missed that, thanks.